src/server/api/routers/question.ts

```diff
@@ -41,50 +41,63 @@ export const questionRouter = createTRPCRouter({
     });
   }),
   getAllReducedInfinite: publicProcedure
-    .input(z.object({ cursor: z.string().optional(), limit: z.number().min(1).max(100).nullable(), titleFilter: z.string().optional() }))
-    .query(
-      async ({input, ctx}) => {
-        const limit = input.limit ?? 50;
-        const totalCount = await ctx.prismaMongo.question.count({
-          where: input.titleFilter ? {
-            OR: [
-              {
-                title: {
-                  contains: input.titleFilter
-                },
-              },
-              {
-                body: {
-                  contains: input.titleFilter
-                }
-              }
-            ]
-          } : undefined,
-        });
-        const items = await ctx.prismaMongo.question.findMany({
-          take: limit + 1,
-          where: input.titleFilter ? {
-            title: {
-              contains: input.titleFilter
+    .input(
+      z.object({
+        cursor: z.string().optional(),
+        limit: z.number().min(1).max(100).optional(),
+        titleFilter: z.string().optional(),
+        backwards: z.boolean().optional(),
+      }),
+    )
+    .query(async ({ input, ctx }) => {
+      const { limit = 50, titleFilter, backwards, cursor } = input;
+      const totalCount = await ctx.prismaMongo.question.count({
+        where: titleFilter
+          ? {
+              OR: [
+                { title: { contains: titleFilter } },
+                { body: { contains: titleFilter } },
+              ],
             }
-          } : undefined,
-          cursor: input.cursor ? {
-            title: input.cursor
-          } : undefined,
-          orderBy: {
-            title: 'asc'
-          }
-        });
-        let nextCursor: string | undefined = undefined;
-        if (items.length === limit + 1) {
-          const top = items.pop();
-          nextCursor = top!.title;
-        }
-        return {
-          items, nextCursor, totalCount
-        };
+          : undefined,
+      });
+      const items = await ctx.prismaMongo.question.findMany({
+        take: backwards ? -limit : limit,
+        skip: cursor ? 1 : 0,
+        where: titleFilter
+          ? {
+              OR: [
+                { title: { contains: titleFilter } },
+                { body: { contains: titleFilter } },
+              ],
+            }
+          : undefined,
+        cursor: cursor ? { title: cursor } : undefined,
+        orderBy: {
+          title: "asc",
+        },
+      });
+      const hasNext = await ctx.prismaMongo.question.count({
```

Comment:
Should prob put next cursor generation here, before any manipulation with has next

Reply:
Actually now that I think about it, this is not needed? we can simply use the total count to determine whether we have next

Reply:
@Gabau Not possible as we are using cursor-based instead of offset-based pagination. The latter does not scale.

Reply:
Using the context of useInfiniteQuery, it's possible if we cache the previous queries. But i think the simplest fix is https://github.com/CS3219-AY2324S1/ay2324s1-course-assessment-g39/pull/94#discussion_r1382402520

Reply:
Cause the current code breaks the pagination on the collab page which uses useInfiniteQuery